src/D2L.Security.OAuth2/OAuth2Configuration.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+
+namespace D2L.Security.OAuth2 {
+	public sealed class OAuth2Configuration {
+		/// <summary>
+		/// When a new key is created we try to avoid using it immediately so
+		/// that caches can expire.
+		/// </summary>
+		public TimeSpan KeyTimeUntilUse { get; init; }
+			= TimeSpan.FromHours( 1 );
+
+		/// <summary>
+		/// We rotate keys before they expire so that any tokens signed by the
+		/// key will become invalid before it expires.
+		/// </summary>
+		public TimeSpan KeyRotationBuffer { get; init; }
+			= TimeSpan.FromDays( 7 );
+
+		/// <summary>
+		/// The total lifetime keys we generate.
+		/// </summary>
+		public TimeSpan KeyLifetime { get; init; }
+			= TimeSpan.FromDays( 30 );

```

Comment:
These defaults are what I was thinking of for the auth service and eventually the LMS.